### PR TITLE
[SPARK-37967][SQL] Literal.create support ObjectType

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Currently , ConstantFolding use `Literal.create()` to represent the result of constant.
If the data type is ObjectType, such as `ObjectType(classOf[UTF8String])`, 
current `Literal.create` will first convert `UTF8String`  to `String`, then failed to create ObjectType literal.
this pr to fix this issue.

### Why are the changes needed?
Fix bug

### Does this PR introduce _any_ user-facing change?
User can create ObjectType literal  use `Literal.create()` more safety.

### How was this patch tested?
Added UT

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan, I think we can merge this before  https://github.com/apache/spark/pull/35207

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala [hunk 1]
```diff
@@ -158,6 +158,7 @@ object Literal {
         Literal(CatalystTypeConverters.createToCatalystConverter(dataType)(v), dataType)
       case _: DayTimeIntervalType if v.isInstanceOf[Duration] =>
         Literal(CatalystTypeConverters.createToCatalystConverter(dataType)(v), dataType)
+      case _: ObjectType => Literal(v, dataType)
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala:161]: shall we do some assertion here? e.g. `v.getClass` should be or a sub-class of `ObjectType.cls`

Comment (AngersZhuuuu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala:161]: > shall we do some assertion here? e.g. `v.getClass` should be or a sub-class of `ObjectType.cls`

Already have a check in `Literal.validateLiteralValue(value, dataType)` when constructing Literal

**Review by cloud-fan — APPROVED** [review]

**Comment by AngersZhuuuu** [comment]
ping @cloud-fan GA passed

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**